src/Var.cc

```diff
@@ -701,8 +701,20 @@ TraversalCode OuterIDBindingFinder::PostExpr(const Expr* expr)
 	return TC_CONTINUE;
 	}
 
+static bool duplicate_ASTs = getenv("ZEEK_DUPLICATE_ASTS");
+
 void end_func(StmtPtr body)
 	{
+	if ( duplicate_ASTs && reporter->Errors() == 0 )
+		// Only try duplication in the absence of errors.  If errors
+		// have occurred, they can be re-generated during the
+		// duplication process, leading to regression failures due
+		// to duplicated error messages.
+		//
+		// We duplicate twice to make sure that the AST produced
+		// by duplicating can itself be correctly duplicated.
+		body = body->Duplicate()->Duplicate();
```

Comment:
A double-duplication seems more like a single test-case than something that has to always be done in reality.  Maybe that's exactly the thinking here since AST-duplication would essentially be unused at this point except as a hypothetical `btest -a AST-dup` run, but seems like an inefficiency that's easy to accidentally leave in later on.

Reply:
> A double-duplication seems more like a single test-case than something that has to always be done in reality

Yep.  But if it succeeds then you know that a single duplication will succeed.

> Maybe that's exactly the thinking here since AST-duplication would essentially be unused at this point except as a hypothetical `btest -a AST-dup` run

Right.

> seems like an inefficiency that's easy to accidentally leave in later on

You can only get to that code path by setting a flag that's off by default, and the use of double-duplicating is documenting, so IMHO it's safe enough in that regard.  That said, all I really want at this point is a prospect of moving forward with the whole-lotta-changes to come ...